packages/cli/src/cmds/agentInstaller/agentProcedure.ts

```diff
@@ -108,6 +106,151 @@ export default abstract class AgentProcedure {
     return validationResult;
   }
 
+  async filesLocallyModified(): Promise<string[]> {
+    let files: string[] = [];
+
+    let gitStatus;
+    try {
+      const stdout = runSync(new CommandStruct('git', ['status', '-s'], this.installer.path));
+      if (stdout.length > 0) {
+        const lines = stdout.split('\n');
+        for (const line of lines) {
+          gitStatus = line.substring(0, 2);
+          const file = line.substring(3);
+          if (gitStatus[0] !== '?') {
+            files.push(file);
+          }
+        }
+      } else {
+        gitStatus = '[no status]';
+      }
+    } catch (e) {
+      const gitError = (e as Error).message.split('\n')[1];
+      gitStatus = `[git status failed, ${gitError}]`;
+    }
+
+    return files;
+  }
+
+  async gitStatus(file: string): Promise<string> {
+    let gitStatus;
+    try {
+      const stdout = runSync(new CommandStruct('git', ['status', '-s', file], this.installer.path));
```

Comment:
Since you are calling and parsing `git status` twice in this PR, you may want the same `gitStatus` function to be used in both cases, instead of having two places where you call and parse `git status`. The `file` argument can be optional.

Reply:
Refactored with `file` optional.

I had followed the pattern for [gitRemote](https://github.com/getappmap/appmap-js/pull/844/files#diff-f040a04bb00a8ecc8c0bedf2772c33579af265dbcdeb56e5d69ceadb4f721097L32) from line 32.  May want to refactor `gitRemote`.